docs/api-reference.md

```diff
@@ -281,6 +292,16 @@ end
 
 Setting a field in the state to `Roact.None` will clear it from the state. This is the only way to remove a field from a component's state!
 
+!!! warning
+	`setState` can be called from anywhere **except**:
+
+	* Lifecycle hooks: `willUnmount`
+	* Pure functions: `render`, `shouldUpdate`
+
+	Calling `setState` inside of `init` or `willUpdate` has different behavior in other places. Because Roact is already going to render or update a component in these cases, the update will be replaced instead of another update happening.
```

Comment:
I don't like the "in other places" part (also, you're missing a "than").  It might be fine to just say "...has special behavior." What do you think?